### PR TITLE
Add e2e test in bundles

### PR DESCRIPTION
**Pull request by ktsakalozos**

### CODE REVIEW

**Comment by mbruzek** [comment]
We need the tests directory copied to the resulting bundles. @wwwtyro Can you help with this?

**Comment by ktsakalozos** [comment]
I think I am doing that here: https://github.com/juju-solutions/bundle-canonical-kubernetes/pull/228/files#diff-94377c156735b39dfa4ac607234cb87cR189 but i could be wrong!

**Comment by wwwtyro** [comment]
@mbruzek lgtm :+1:

**Comment by wwwtyro** [comment]
Looks correct to me! If it's been tested, I'm good.

**Comment by mbruzek** [comment]
@ktsakalozos These are 2 separate subjects and should be in different PRs. I want to add the copy tree. But you mentioned that 30 test is unfinished and I would like to review that one separately.

File: tests/30-kubernetes-e2e-check.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/python3
```
Comment (mbruzek) [inline tests/30-kubernetes-e2e-check.py]: I have heard `#!/usr/bin/env python3` is more portable. This is not a blocking comment, just want to be consistent with other stuff I have seen out of the python tools.

File: tests/30-kubernetes-e2e-check.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                                'kubernetes-master:kube-api-endpoint')
+
+    def test_e2e(self):
+        '''Trigger e2e tests'''
+        args = {'skip': '\[(Flaky|Slow|Feature:.*)\]'}
+        action_id = self.e2e.run_action('test', args)
+        outcome = self.deployment.action_fetch(action_id,
+                                               timeout=7200,
+                                               raise_on_timeout=True,
+                                               full_output=True )
```
Comment (mbruzek) [inline tests/30-kubernetes-e2e-check.py]: This line fails proof. 

`tests/30-kubernetes-e2e-check.py:59:64: E202 whitespace before ')'`

File: tests/tests.yaml [hunk 3]
```diff
@@ -13,4 +13,5 @@ packages:
   - amulet
```
Comment (mbruzek) [inline tests/tests.yaml]: Do we actually need amulet and juju-deployer here any longer? @ktsakalozos I know you didn't change this but might know better than us about the requirements for packages here.

File: tests/30-kubernetes-e2e-check.py [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+                              'easyrsa:client')
+        cls.deployment.relate('kubernetes-e2e:kubernetes-master',
+                              'kubernetes-master:kube-api-endpoint')
+        # Wait for the system to settle down.
+        application_messages = {'kubernetes-worker':
+                                'Kubernetes worker running.',
+                                'kubernetes-e2e':
+                                'Ready to test.'}
+        cls.deployment.sentry.wait_for_messages(application_messages,
+                                                timeout=1800)
```
Comment (mbruzek) [inline tests/30-kubernetes-e2e-check.py]: Why don't you use SECONDS_TO_WAIT here as well?

File: tests/30-kubernetes-e2e-check.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+        cls.e2e = cls.deployment.sentry['kubernetes-e2e'][0]
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.deployment.unrelate('kubernetes-e2e:certificates',
```
Comment (mbruzek) [inline tests/30-kubernetes-e2e-check.py]: @ktsakalozos explained this removal of the relation to me here, and I really like that feature. Needs a docstring for this method, but other than that +1

File: tests/30-kubernetes-e2e-check.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+
+SECONDS_TO_WAIT = 1800
+
+
+class E2eIntegrationTest(unittest.TestCase):
+    bundle_file = os.path.join(os.path.dirname(__file__), '..', 'bundle.yaml')
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (mbruzek) [inline tests/30-kubernetes-e2e-check.py]: Methods without docstrings make kittens (and other software developers) cry.

File: tests/tests.yaml (hunk 3 above)
Comment (ktsakalozos) [inline tests/tests.yaml]: I think we still need this. Haven't heard anything.

File: tests/30-kubernetes-e2e-check.py (hunk 6 above)
Comment (ktsakalozos) [inline tests/30-kubernetes-e2e-check.py]: Done

**Comment by ktsakalozos** [comment]
Removed the copy of tests from this PR and created a separate one here: 
https://github.com/juju-solutions/bundle-canonical-kubernetes/pull/229

**Comment by mbruzek** [comment]
+1 Thanks for addressing most of my comments.

**Pull request merged**